registry/tracecat_registry/integrations/microsoft_teams.py

```diff
@@ -0,0 +1,795 @@
+                    "roles": ["owner"],
+                    "user@odata.bind": f"https://graph.microsoft.com/beta/users('{user_id}')",
+                }
+            )
+
+        payload = {
+            "@odata.type": "#Microsoft.Graph.channel",
+            "membershipType": "private",
+            "displayName": display_name,
+            "description": description or "",
+            "members": members,
+        }
+    else:
+        payload = {
+            "displayName": display_name,
+            "description": description or "",
+            "membershipType": "standard",
+        }
+
+    async with httpx.AsyncClient() as client:
+        response = await client.post(url, headers=headers, json=payload)
+        response.raise_for_status()
+        return response.json()
+
+
+@registry.register(
+    default_title="List channel messages",
+    description="Retrieve the list of messages (without the replies) in a channel of a team.",
+    display_group="Microsoft Teams",
+    doc_url="https://learn.microsoft.com/en-us/graph/api/channel-list-messages?view=graph-rest-beta&tabs=python",
+    namespace="tools.microsoft_teams",
+    secrets=[microsoft_teams_oauth_secret],
+)
+async def list_channel_messages(
+    team_id: Annotated[
+        str,
+        Doc("The ID of the team containing the channel."),
+    ],
+    channel_id: Annotated[
+        str,
+        Doc("The ID of the channel to list messages from."),
+    ],
+    top: Annotated[
+        int | None,
+        Doc("Number of messages to return per page (default 20, max 50)."),
+    ] = None,
+    expand_replies: Annotated[
+        bool,
+        Doc("Whether to expand replies for each message."),
+    ] = False,
+) -> dict[str, str]:
+    """List messages from a Teams channel.
+
+    Note: This API requires ChannelMessage.Read.All or ChannelMessage.Read.Group permissions.
+    """
+    token = secrets.get("MICROSOFT_TEAMS_ACCESS_TOKEN")
+
+    headers = {"Authorization": f"Bearer {token}"}
+
+    url = f"https://graph.microsoft.com/beta/teams/{team_id}/channels/{channel_id}/messages"
+
+    params = {}
+    if top:
+        if top > 50:
+            raise ValueError("Top parameter cannot exceed 50 messages per page")
+        params["$top"] = top
+
+    if expand_replies:
+        params["$expand"] = "replies"
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(url, headers=headers, params=params)
+        response.raise_for_status()
+        return response.json()
+
+
+@registry.register(
+    default_title="Get user ID by email",
+    description="Get a user's ID by searching for their email address in mail or userPrincipalName Docs.",
+    display_group="Microsoft Teams",
+    doc_url="https://learn.microsoft.com/en-us/graph/api/user-list?view=graph-rest-beta&tabs=http",
+    namespace="tools.microsoft_teams",
+    secrets=[microsoft_teams_oauth_secret],
+)
+async def get_user_id_by_email(
```

Comment:
this action isn't teams related. should it go under `microsoft_entra`? cc @topher-lo